lib/stylelint/plugin.rb

```diff
@@ -1,34 +1,117 @@
 # frozen_string_literal: true
 
+require "mkmf"
+require "json"
+
 module Danger
-  # This is your plugin class. Any attributes or methods you expose here will
-  # be available from within your Dangerfile.
-  #
-  # To be published on the Danger plugins site, you will need to have
-  # the public interface documented. Danger uses [YARD](http://yardoc.org/)
-  # for generating documentation from your plugin source, and you can verify
-  # by running `danger plugins lint` or `bundle exec rake spec`.
+  # Lint javascript files using [stylelint](http://stylelint.io/).
+  # Results are send as inline comment.
   #
-  # You should replace these comments with a public description of your library.
+  # @example Run stylelint with changed files only
   #
-  # @example Ensure people are well warned about merging on Mondays
+  #          stylelint.filtering = true
+  #          stylelint.lint
   #
-  #          my_plugin.warn_on_mondays
-  #
-  # @see  Rossukhon Leagmongkol/danger-stylelint
-  # @tags monday, weekends, time, rattata
+  # @see  nimblehq/danger-stylelint
+  # @tags lint, stylelint, css, scss, sass
   #
   class DangerStylelint < Plugin
-    # An attribute that you can read/write from your Dangerfile
+    DEFAULT_BIN_PATH = "./node_modules/.bin/stylelint"
+
+    # An path to stylelint's config file
+    # @return [String]
+    attr_accessor :config_file
+
+    # Enable filtering
+    # Only show messages within changed files.
+    # @return [Boolean]
+    attr_accessor :filtering
```

Comment:
I'm wondering if we could change this to `changes_only` or something like that 🤔 

So when the user configures the option it's more descriptive of what it does

```ruby
stylelint.changes_only = true
```

Reply:
sure,  it's a good idea, done in 4f636e6f6e7e07faf138ee91211bdca380637e61!